### DISCUSSION TITLE
nodejs-validation-performance/

### DISCUSSION THREAD

**Discussion by giscus** [JHyeok/JHyeok.com#54] [Comments]
# nodejs-validation-performance/

저는 NestJS를 사용하면 유효성 검사와 직렬화 라이브러리에 class-validator와 class-transformer를 사용합니다. 반면에 Express를 사용하는 경우에는 Joi를 사용하고 있는데요. 하지만 Joi…

https://jhyeok.com/nodejs-validation-performance/

**Comment by samchon** [comment]
모두 동일한 속도가 나온 이유는 DB I/O에서 병목이 걸렸기 때문입니다.

그리고 느린 validation 및 serialization은 request DTO보단, response DTO에서 서버 성능에 끼치는 해악이 더 큽니다. 마지막으로 이런 벤치마크 실험을 할 때는 DTO를 하나만 두고 할 게 아니라, 다양한 종류를 두고 실험해야 합니다.

실제로 DB I/O가 없고, 다양한 구조의 DTO를 실험했을 때는 그 결과가 상이합니다.

- 서버 사이드 벤치마크: https://github.com/samchon/nestia/…%20Core(TM)%20i5-1135G7%20%40%202.40GHz
- 순수 Validation 속도 벤치마크: https://github.com/samchon/typia/…%20Core(TM)%20i5-1135G7%20%40%202.40GHz
